applications/FluidDynamicsApplication/custom_utilities/cfd_utilities.cpp

```diff
@@ -0,0 +1,429 @@
+//
+
+// System includes
+#include <functional>
+#include <limits>
+
+// External includes
+
+// Project includes
+#include "utilities/math_utils.h"
+#include "utilities/parallel_utilities.h"
+#include "utilities/variable_utils.h"
+
+// Application includes
+#include "fluid_dynamics_application_variables.h"
+
+// Include base h
+#include "cfd_utilities.h"
+
+namespace Kratos
+{
+namespace CFDUtilities
+{
+void CalculateConditionGeometryData(const GeometryType& rGeometry,
+                                    const GeometryData::IntegrationMethod& rIntegrationMethod,
+                                    Vector& rGaussWeights,
+                                    Matrix& rNContainer)
+{
+    const GeometryType::IntegrationPointsArrayType& integration_points =
+        rGeometry.IntegrationPoints(rIntegrationMethod);
+
+    const std::size_t number_of_integration_points = integration_points.size();
+    const int dimension = rGeometry.WorkingSpaceDimension();
+    const double domain_size = rGeometry.DomainSize();
+
+    if (rGaussWeights.size() != number_of_integration_points)
+    {
+        rGaussWeights.resize(number_of_integration_points, false);
+    }
+
+    rNContainer = rGeometry.ShapeFunctionsValues(rIntegrationMethod);
+
+    // CAUTION: "Jacobian" is 2.0*A for triangles but 0.5*A for lines
+    double det_J = (dimension == 2) ? 0.5 * domain_size : 2.0 * domain_size;
+
+    for (unsigned int g = 0; g < number_of_integration_points; g++)
+    {
+        rGaussWeights[g] = det_J * integration_points[g].Weight();
+    }
+}
+
+template <>
+void CalculateConditionNormal<2>(array_1d<double, 3>& rNormal, const ConditionType& rCondition)
+{
+    const GeometryType& pGeometry = rCondition.GetGeometry();
+
+    rNormal[0] = pGeometry[1].Y() - pGeometry[0].Y();
+    rNormal[1] = -(pGeometry[1].X() - pGeometry[0].X());
+    rNormal[2] = 0.00;
+}
+
+template <>
+void CalculateConditionNormal<3>(array_1d<double, 3>& rNormal, const ConditionType& rCondition)
+{
+    const GeometryType& pGeometry = rCondition.GetGeometry();
+
+    array_1d<double, 3> v1, v2;
+    v1[0] = pGeometry[1].X() - pGeometry[0].X();
+    v1[1] = pGeometry[1].Y() - pGeometry[0].Y();
+    v1[2] = pGeometry[1].Z() - pGeometry[0].Z();
+
+    v2[0] = pGeometry[2].X() - pGeometry[0].X();
+    v2[1] = pGeometry[2].Y() - pGeometry[0].Y();
+    v2[2] = pGeometry[2].Z() - pGeometry[0].Z();
+
+    MathUtils<double>::CrossProduct(rNormal, v1, v2);
+    rNormal *= 0.5;
+}
+
+double CalculateConditionWallHeight(const ConditionType& rCondition,
+                                    const array_1d<double, 3>& rNormal)
+{
+    KRATOS_TRY
+
+    array_1d<double, 3> normal = rNormal / norm_2(rNormal);
+
+    const ElementType& r_parent_element = rCondition.GetValue(NEIGHBOUR_ELEMENTS)[0];
+
+    const GeometryType& r_parent_geometry = r_parent_element.GetGeometry();
+    const GeometryType& r_condition_geometry = rCondition.GetGeometry();
```

Comment:
I'd use the `auto` types in here.

Reply:
I was thinking of using actual types to reduce compilation time. do you think it doesn't affect much if I use auto?

Reply:
I don't think it affects so much and I found it worth it for the code maintenance.